include/libtwofloat/arithmetics/double-word-arithmetic.hpp

```diff
@@ -233,5 +233,628 @@ inline two<T> div(const two<T> &x, const two<T> &y) {
+  const T* local_ptr_inv_fact;
+  T zeropointzero;
+  if constexpr (std::is_same_v<T, float>) {
+    pointfive = 0.5f;
+    local_eps = fp32_eps;
+    onepointzero = 1.0f;
+    local_ptr_inv_fact = &fp32_inv_fact[0][0];
+    zeropointzero = 0.0f;
+  } else if constexpr (std::is_same_v<T, double>) {
+    pointfive = 0.5;
+    local_eps = fp64_eps;
+    onepointzero = 1.0;
+    local_ptr_inv_fact = &fp64_inv_fact[0][0];
+    zeropointzero = 0.0;
+  } else {
+    std::cerr << "Other types not supported\n";
+  }
+
+  const T thresh = pointfive * local_eps;
+
+  two<T> r, s, t, x;
+
+  if (input.eval() == zeropointzero) {
+    T hi, lo;
+    qd::split(onepointzero, hi, lo);
+    return two<T>{hi, lo};
+  }
+
+  two<T> temp = sqr(input);
+  x = two<T>{-temp.h, -temp.l};
+  r = x;
+  s = add(onepointzero, mul_pwr2(r, pointfive));
+  int i = 1;
+  do {
+    r = mul<doubleword::Mode::Accurate, true>(r, x);
+    t = mul<doubleword::Mode::Accurate, true>(r, two<T>{(local_ptr_inv_fact+i)[0], (local_ptr_inv_fact+i)[1]});
+    s = add<doubleword::Mode::Accurate>(s, t);
+    i += 2;
+  } while (i < n_inv_fact && std::abs(to_double(t)) > thresh);
+
+  return s;
+}
+
+// Reference: QD / dd_real.cpp
+/* Computes the square root of the double-double number dd.
+   NOTE: dd must be a non-negative number. */
+template <typename T>
+two<T> sqrt(const two<T> &input) {
+  /* Strategy:  Use Karp's trick:  if x is an approximation
+     to sqrt(a), then
+
+        sqrt(a) = a*x + [a - (a*x)^2] * x / 2   (approx)
+
+     The approximation is accurate to twice the accuracy of x.
+     Also, the multiplication (a*x) and [-]*x can be done with
+     only half the precision.
+  */
+
+  T zeropointzero, onepointzero, pointfive, local_nan;
+  if constexpr (std::is_same_v<T, float>) {
+    onepointzero = 1.0f;
+    zeropointzero = 0.0f;
+    pointfive = 0.5f;
+    local_nan = fp32_nan;
+  } else if constexpr (std::is_same_v<T, double>) {
+    onepointzero = 1.0;
+    zeropointzero = 0.0;
+    pointfive = 0.5;
+    local_nan = fp64_nan;
+  } else {
+    std::cerr << "Other types not supported\n";
+  }
+
+  if (input.eval() == zeropointzero) {
+    T hi, lo;
+    qd::split(zeropointzero, hi, lo);
+    return two<T>{hi, lo};
+  }
+
+  // if (input.is_negative()) {
+  if (input.h < zeropointzero) {
+    std::cerr << "Negative argument.\n";
+    return two<T>{local_nan, local_nan};
+  }
+
+  T x = onepointzero / std::sqrt(input.h);
+  T ax = input.h * x;
+  two<T> temp = sub<doubleword::Mode::Accurate>(input, dd_real::sqr(ax));
+  return dd_real::add(ax, temp.h * x * pointfive);
+}
+
+// Reference: QD / dd_real.cpp
+template <typename T>
+static void sincos_taylor(const two<T> &input, two<T> &sin_a, two<T> &cos_a) {
+
+  T zeropointzero, onepointzero;
+  if constexpr (std::is_same_v<T, float>) {
+    zeropointzero = 0.0f;
+    onepointzero = 1.0f;
+  } else if constexpr (std::is_same_v<T, double>) {
+    zeropointzero = 0.0;
+    onepointzero = 1.0;
```

Comment:
Just assign 0 and and 1, the compiler will handle the casts for us

Reply:
Addressed in [9e1e0be](https://github.com/esa-tu-darmstadt/twofloat/pull/1/commits/9e1e0be7403cab689209d28400aef46366224a6d).